pkg/api/api.go

```diff
@@ -0,0 +1,201 @@
+	kindMapping := map[string]schema.GroupVersionResource{
+		"ConfigMap":  {Group: "", Version: "v1", Resource: "configmaps"},
+		"Deployment": {Group: "apps", Version: "v1", Resource: "deployments"},
+		"Job":        {Group: "batch", Version: "v1", Resource: "jobs"},
+		"Pod":        {Group: "", Version: "v1", Resource: "pods"},
+		"Namespace":  {Group: "", Version: "v1", Resource: "namespaces"},
+		"Node":       {Group: "", Version: "v1", Resource: "nodes"},
+		"Secret":     {Group: "", Version: "v1", Resource: "secrets"},
+		"Service":    {Group: "", Version: "v1", Resource: "services"},
+	}
+
+	gvk, found := kindMapping[kind]
+	if !found {
+		return schema.GroupVersionResource{}, fmt.Errorf("unknown kind: '%s'", kind)
+	}
+	return gvk, nil
+}
+
+// Kubernetes defines an interface that extends kubernetes interface[k8s.io/client-go/kubernetes.Interface] adding
+// generic functions that operate on any kind of object
+type Kubernetes interface {
+	Apply(manifest string) error
+	Create(obj map[string]interface{}) (map[string]interface{}, error)
+	Get(kind string, name string, namespace string) (map[string]interface{}, error)
+	List(kind string, namespace string) ([]map[string]interface{}, error)
+	Delete(kind string, name string, namespace string) error
+}
+
+// KubernetesConfig defines the configuration for creating a Kubernetes instance
+type KubernetesConfig struct {
+	// Context for executing kubernetes operations
+	Context context.Context
+	// kubernetes rest config
+	Config *rest.Config
+	// Client is a pre-configured dynamic client. If provided, the rest config is not used
+	Client dynamic.Interface
+}
+
+// kubernetes Holds the reference to the helpers for interacting with kubernetes
+type kubernetes struct {
+	ctx        context.Context
+	client     dynamic.Interface
+	serializer runtime.Serializer
+}
+
+// NewFromConfig returns a Kubernetes instance
+func NewFromConfig(c KubernetesConfig) (Kubernetes, error) {
+	client := c.Client
+	var err error
+	if client == nil {
+		client, err = dynamic.NewForConfig(c.Config)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	ctx := c.Context
+	if ctx == nil {
+		ctx = context.TODO()
+	}
+
+	return &kubernetes{
+		ctx:        ctx,
+		client:     client,
+		serializer: yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme),
+	}, nil
+}
+
+// Apply creates a resource in a kubernetes cluster from a yaml manifest
+func (k *kubernetes) Apply(manifest string) error {
+	uObj := &unstructured.Unstructured{}
+	_, gvk, err := k.serializer.Decode([]byte(manifest), nil, uObj)
+	if err != nil {
+		return err
+	}
+	resource, err := knownKinds(gvk.Kind)
+	if err != nil {
+		return err
+	}
+
+	namespace := uObj.GetNamespace()
+	if namespace == "" {
+		namespace = "default"
+	}
+
+	_, err = k.client.Resource(resource).
+		Namespace(namespace).
+		Create(
+			k.ctx,
+			uObj,
+			metav1.CreateOptions{},
+		)
+	return err
+}
+
+// Create creates a resource in a kubernetes cluster from a yaml manifest
```

Comment:
```suggestion
// Create creates a resource in a Kubernetes cluster from a YAML manifest
```

Reply:
The comment is actually wrong as this method does not receive a YAML but an object with the resource specification